### PR TITLE
Authentication Project

### PR DESCRIPTION
**Pull request by Paul-Clue**
## What We Did:
- We created a SQL database.
- We made it so that when the user is logged out, the user cannot see who owns any given post in the data base.
- We implemented authentication using devise.

### CODE REVIEW

**Review by peterrobert — CHANGES_REQUESTED** [review]
### INVALID 🛑 
Hello @Paul-Clue  and @carloso0114  👋🏿  Good work submitting your project for a review. This shows great effort. However, for your project to be reviewed, it must meet the basic minimum requirements. Unfortunately, your project lucks one of these requirements kindly check below ⬇️ 

- [x] Kindly add a descriptive README.md. This will make your project more professional. Kindly follow [this](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/) link to learn more about the minimum requirements.

descriptive_readme


But no worries, just add a descriptive readme file and submit your project for another review.

### if you have any question, kindly reach me via slack @peterrobert

### HAPPY CODING

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @Paul-Clue and @carloso0114 :clap: :clap:
#### Good job so far :+1: However, some requirements are missing/not implemented correctly. Check my comments below :point_down:

#### 1) Routes:
 - The posts routes should be `resources :posts, only: [:new, :create, :index]`

#### 2) PostController:
 - it uses `before_action` to restrict access to the `:new` and `:create` methods
 - `new` action method: initializes an empty post `@post = Post.new`

#### _Reviewed by Temitayo Adeleke_

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
# **Changes Required 🛑♻**

<hr>

Hello there, great job working on this project and on the required changes. However, there are still a few more issues that need to be fixed in order to get this PR approved. :)
Please keep up the great work :D

<hr>

- [x] Since you modified your routes as requested by the previous reviewer to only match the new, create, and index methods. You should also modify your posts controller and views to remove any related code to the other methods (show, edit, update, and destroy) that won't be working if they're not declared in the routes resources. Please fix this issue.
- [ ] Following the previous point. When I create a new post, I get this error ```undefined method `post_path'``` due to the missing resource show in the routes. Kindly modify the views due to the current resources in your routes file.
- [ ] In index file. Please move any logic code into the helpers. There should not be any logic code in the views).

<hr>

Please do the required changes, and ask for another review.
Good luck and happy coding 👨‍💻💻

**Review by mke2111 — CHANGES_REQUESTED** [review]
# **STATUS: CHANGES REQUIRED 🔴 ⚠️** 

## Hey @Paul-Clue & @carloso0114, 👋🏽 
 **Thank you for the good job you got going on. The app runs almost just fine.**

## REQUIRED CHANGES

- [x] When I start the server, this is the page I'm presented with and there is no way to navigate to sign up or log in. Give this a fix.
Screenshot_2021-03-09_03-56-19

- [x] Also add navigation buttons/links especially `back` and `sign_out` once a new post is created. You can choose to add
- Sign_out
- New Post
- Back
- All posts
- My posts.
- This is what I see after creating a post.
Screenshot_2021-03-09_04-12-32


- [ ] Right now, I cannot sign_out since there is no button/link for that on any of the pages. Kindly add that.
- [ ] The application layout has Sign in / Sign out links So they are not manually repeated in every view


**Kindly make all the requested changes and submit another CRR. 👍🏽** 

## Cheers!! 🍹 & Happy Coding.


## Ping me on slack for a  🆘  **Roy Mukuye** or mention me [@mke2111](https://github.com/mke2111) in this PR and I'll come through for your inquiries.

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1 or /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    # @post = current_user.posts.build
```
Comment (enaburekhan) [inline app/controllers/posts_controller.rb:15]: Remove the commented out code. It makes it difficult to understand the codebase.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:full_moon:
**Hi team!
Great job so far:clap::clap:. The app is working as expected**

**major highlights**

- A user can sign-in and a signed-in user can create a new post
- A user can sign out.

However, we just need to make a few changes to get the project approved:muscle:.

**To improve**

- [ ] PostsController

- it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). We need to create this method in application helper and then require it with the before_action in the postController. I.e `before_action :signed_in_only!, only: [:new, :create]`

- [ ] Remove commented out code in `new` in PostsController as already highlighted in the PR below.

Make the above-required changes and when you are ready, request another review.
Kindly drop your questions :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy coding:blush:
Eric Enaburekhan

**Pull request merged**